src/Components/StepTwo/StepTwo.js

```diff
@@ -1,8 +1,118 @@
-import React, {useState} from 'react';
+import React, {useState, useEffect} from 'react';
 import './StepTwo.scss';
+import axios from 'axios';
+import useFetchForm from '../useFetchForm/useFetchForm';
+import { url2 } from '../../Constants';
 
+const StepTwo = ({prevStep, nextStep, data, setData}) => {
+
+  const {dataStates} = useFetchForm(url2);
+  
+  const [errors, setErrors] = useState({})
+  const [dataCities, setDataCities] = useState([])
+  const [dataPostalCodes, setDataPostalCodes] = useState([])
+  const [tla, setTla] = useState('');
+    
+  useEffect(()=> {
+        if (tla) {
+        axios.get(url2 + "/" + tla + "/cities")
+        .then(res => {
+            setDataCities(res.data)
+        })
+    }
+    },[tla])
+
+    const [inputs, setInputs] = useState({
+      email: data.email,
+      residencyDuration: data.residencyDuration,
+      address: {
+            addressLine1: data.address.addressLine1,
+            addressLine2: data.address.addressLine2,
+            city: data.address.city,
+            state: data.address.state,
+            postalCode: data.address.postalCode,
+      },
+       
+    });
+    
+    useEffect(()=> {
+      if (inputs.address.city) {
+      axios.get(url2 + "/" + tla + "/cities/" + inputs.address.city + "/postalcodes" )
+      .then(res => {
+          setDataPostalCodes(res.data)
+      })
+  }
+  },[inputs.address.city])
+
+ 
+    const handleChange = (e) => {
+        if ( e.target.name === "residencyDuration" || e.target.name ==="email") {
+        e.preventDefault();
+        setInputs((prevState) => ({
+          ...prevState,
+          [e.target.name]: e.target.value,
+        })
+        )
+      }  else {
+          e.preventDefault();
+          setInputs((prevStep) => ({
+              ...prevStep,
+              address: {
+                  ...inputs.address,
+                  [e.target.name]: e.target.value
+              }
+          }))
+          if (e.target.name === "state") {
+            const findTla = dataStates.find((state) => state.name === e.target.value);
+            findTla && setTla(findTla.tla);
+          }
+      }};
+    
+    const onSubmitForm = (e) => {
+        e.preventDefault();
+        const isValid = formValidation()
+        if(isValid) {
+          nextStep(3)
+        }
+      }
+ 
+    const formValidation = () => {
+        const emailErr = {};
+        const residencyDurationErr = {};
+        let isValid = true;
+  
+        if (!inputs.email) {
+          emailErr.email = "Email required!";
+          isValid = false;
+        } else if (!/\S+@\S+\.\S+/.test(inputs.email)) {
```

Comment:
The regex, and in general, this kind of validation function, should be saved to separate file. Maybe some kind of utils or helpers?